commit/chainfee/observation.go

```diff
@@ -39,6 +39,17 @@ func (p *processor) Observation(
 		return Observation{}, err
 	}
 
+	// filter out disabled chains
+	sourceChainsConfig, err := p.ccipReader.GetOffRampSourceChainsConfig(ctx, supportedChains.ToSlice())
+	if err != nil {
+		return Observation{}, fmt.Errorf("get offRamp source chains config: %w", err)
+	}
+	for selector, config := range sourceChainsConfig {
+		if !config.IsEnabled {
+			supportedChains.Remove(selector)
+		}
+	}
```

Comment:
This logic should probably go into `SupportedChains`.
@makramkd do you know if there's a reason we don't check the enabled flag?

Reply:
I think it was an optimization, this lookup will need to be cached since it happens within the `Observation`. OCR3 doesn't like it if a round starts taking too long

Reply:
I'll revert this and see if reports are still getting generated.

Reply:
I remember now, we need https://github.com/smartcontractkit/chainlink-ccip/pull/667 merged first